### PR TITLE
Bugfix/retry if no blackboard

### PR DESCRIPTION
**Pull request by improbable-danny**
Fix for crash in test pipeline where simplayers sometimes are not fully initialized (native)

### CODE REVIEW

File: Game/Source/GDKTestGyms/Private/DeterministicBlackboardValues.cpp [hunk 1]
```diff
@@ -48,7 +48,11 @@ void UDeterministicBlackboardValues::ApplyValues() // Repeats until the Componen
 	if (AIController)
 	{
 		UBlackboardComponent* Blackboard = Cast<UBlackboardComponent>(AIController->GetBlackboardComponent());
-		checkf(Blackboard, TEXT("AI Controller did not have a blackboard %s"), *Controller->GetPawn()->GetName());
+		if (!Blackboard)
```
Comment (m-samiec) [inline Game/Source/GDKTestGyms/Private/DeterministicBlackboardValues.cpp:51]:
```suggestion
		if (Blackboard == nullptr)
```

File: Game/Source/GDKTestGyms/Private/DeterministicBlackboardValues.cpp [hunk 2]
```diff
@@ -48,7 +48,11 @@ void UDeterministicBlackboardValues::ApplyValues() // Repeats until the Componen
...
+		{
+			UE_LOG(LogDeterministicBlackboardValues, Log, TEXT("No blackboard, retrying in 1s. Pawn:%s Controller:%s"), *Pawn->GetName(), Controller ? *Controller->GetName() : TEXT("Unset"));
```
Comment (m-samiec) [inline Game/Source/GDKTestGyms/Private/DeterministicBlackboardValues.cpp:53]: You can use GetNameSafe for these two - ie. GetNameSafe(Pawn), GetNameSafe(Controller)

File: Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp [hunk 3]
```diff
@@ -319,7 +319,14 @@ void ABenchmarkGymGameMode::SpawnNPC(const FVector& SpawnLocation, const FBlackb
 		return;
 	}
 
-	FVector FixedSpawnLocation = SpawnLocation;
+	const float RandomSpawnOffset = 600.0f;
+	FVector RandomOffset = FMath::VRand()*RandomSpawnOffset;
```
Comment (m-samiec) [inline Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp:323]: What's all this about?

Comment (improbable-danny) [reply Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp:323]: Random offset for spawning NPCs - they all spawn on top of each other and on top of the players also. I removed this in a previous PR but it seems to be quite beneficial (the players and npcs flew off into the sky)

Comment (m-samiec) [reply Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp:323]: Feature?

**Pull request merged**